ukat/segment/whole_kidney.py

```diff
@@ -0,0 +1,207 @@
+import csv
+import os
+import nibabel as nib
+import numpy as np
+
+from segment import Tkv
+
+
+class Segmentation(nib.Nifti1Image):
+    """
+    Attributes
+    ----------
+    volumes : dict
+        Total, left and right kidney volumes in milliliters.
+    """
+
+    def __init__(self, pixel_array, affine, post_process=True):
+        """Initialise a whole kidney segmentation class instance.
+
+        Parameters
+        ----------
+        pixel_array : np.ndarray
+            Array containing a T2-weighted FSE image.
+        affine : np.ndarray
+            A matrix giving the relationship between voxel coordinates and
+            world coordinates.
+        post_process : bool, optional
+            Default True
+            Keep only the two largest connected volumes in the mask. Note
+            this may cause issue with subjects that have more or less than
+            two kidneys.
+        """
+
+        super().__init__(pixel_array, affine)
+        self.pixel_array = pixel_array
+        self._nifti = nib.Nifti1Image(self.pixel_array, self.affine)
+        self._seg_obj = Tkv(self._nifti)
+        self._mask = self._seg_obj.get_mask(post_process=post_process)
+        self._kidneys = (self._mask > 0.5) * 1
```

Comment:
In reality, the masks are pretty close to binary anyway with very few voxels between 0.05 and 0.95. Changing the threshold to anything less than 0.95 (or greater than 0.05) would have very little impact on the resulting mask so I think this would just be another parameter to confuse people.
![image](https://user-images.githubusercontent.com/23356429/148954018-82f3b945-5e80-4f3b-b7a3-9d4b620cfa7f.png)

Reply:
I see, thank you so much for the excellent explanation and for the histograms showing this!